### ISSUE TITLE
Basic and JWT security schemas overrides `Authorization` header

### ISSUE THREAD

**Issue by xSAVIKx** [mrin9/RapiDoc#554]
So we're using the following setup:
1. Username + password ("basic") auth in order to get a JWT token for working with other API methods.
2. We use `JWT` token as "API key" header authentication schema.

```
components:
  securitySchemes:
    basic:
      type: http
      scheme: basic
    Authorization:
      type: apiKey
      in: header
      name: Authorization 
paths:
  /cases/auth:
    get:
      security:
        - basic: [ ]
  /cases/doSmth:
    post:
      security:
        - Authorization: [ ]
```

Now what happens, when we use our `auth` endpoint, we use the programmatic API to set the `Authorization` API key. From this moment, we have both `basic` and `apiKey` auth set.

And now if we use the `auth` endpoint again, the JWT token is sent to the endpoint instead of the basic auth.

Just FYI. We have also Swagger UI configured, so it behaves properly in this case. If the schema defines the required security component, only the one noted in the schema is applied.

**Comment by xSAVIKx** [comment]
I've tested the setup and it looks like we see the same behavior with both `9.0.0` and `9.1.0` versions. While we're using react app, I'm not sure how to test `9.1.1-beta` with our setup.

**Comment by mrin9** [comment]
> I'm not sure how to test 9.1.1-beta with our setup.

download `9.1.1-beta` from  https://raw.githubusercontent.com/mrin9/RapiDoc/master/dist/rapidoc-min.js and store it somewhere locally on your server

then in your html where you are using `<rapi-doc>` element use the downloaded `rapidoc-min.js` in the `<script>` tag
```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">

    <!-- this is the rapidoc beta check out the first line of rapidoc-min.js to find the version --> 
    <script type="module" src="path/to/downloaded/rapidoc-min.js"></script>

  </head>
  <body>
    <rapi-dpc spec-url="..."> </rapi-doc>
  </body>
</html>

```

**Comment by xSAVIKx** [comment]
Yeah, as I mentioned we're using React components and all dependencies are grabbed from NPM and transpiled. I'll to test with the beta version though or maybe wait for it to be released.

**Comment by mrin9** [comment]
`9.1.2` is released now ... hope it addresses your issues

**Comment by xSAVIKx** [comment]
@mrin9 unfortunately the issue is still present in the latest 9.1.2 version.

It just looks like the "apiKey" authentication always overrides the "basic" authentication details. And it's not clear why both are applied at the same time for an endpoint that clearly states that only "basic" auth is supported.

**Comment by mrin9** [comment]
thanks for testing this out for us... i will further investigate on this one. we have the release window open till sept-1st week so hopefully address it before that

**Comment by mrin9** [comment]
I created a test case as you specified above two security schemes applied to two different apis,
and in my test, it works as expected. 
you can provide these values either through RapiDoc's default Ui or using RapiDoc's method and you will see it is sending the correct header and value in the browser's devtool.

below is the test-case that you can try and test at ur end
https://mrin9.github.io/RapiDoc/examples/auth-test1.html

**Comment by xSAVIKx** [comment]
@mrin9 I've just updated the code to use the new `setApiKey` functionality while https://github.com/mrin9/RapiDoc/issues/492 is now resolved.

Now the described case works perfectly fine. I believe it wasn't working due to some conflicts with the usage of component params. It kinda always showed me that 2 auth schemas are applied to the endpoint and was picking API key always.

So the `v9.1.2` + `setApiKey` works great.

Thanks a lot for checking this out!
